src/components/ErrorModal.vue

```diff
@@ -0,0 +1,267 @@
+<template>
+  <div
+    class="error-modal"
+    @click="resolve"
+  >
+    <div class="modal-body" @click.stop>
+      <IconClose
+        class="close-modal"
+        @click="resolve"
+      />
+      <template v-if="step === 1">
+        <FailureIcon />
+        <h1>{{ $t('reportBug.Title') }}</h1>
+        <h2>{{ $t('reportBug.SubTitle') }}</h2>
+        <p class="description">
+          {{ $t('reportBug.Description') }}
+        </p>
+        <ButtonPlain
+          class="detailsButton"
+          @click="toggleDetails"
+        >
+          <IconEye />
+          {{ $t('reportBug.ViewDetails') }}
+        </ButtonPlain>
+        <p v-show="details">
+          {{ JSON.stringify(error) }}
+        </p>
+        <h2 class="reportTitle">
+          {{ $t('reportBug.DescriptionTitle') }}
+        </h2>
+        <textarea
+          v-model="report"
+          class="report-text"
+          :placeholder="$t('reportBug.Placeholder')"
+        />
+        <Button
+          class="cancel"
+          @click="resolve"
+        >
+          {{ $t('reportBug.Cancel') }}
+        </Button>
+        <Button
+          @click="send"
+        >
+          {{ $t('reportBug.Send') }}
+        </Button>
+      </template>
+      <template v-else>
+        <IconOK />
+        <h1 class="successTitle">
+          {{ $t('reportBug.Success.Title') }}
+        </h1>
+        <h2 class="successSubTitle">
+          {{ $t('reportBug.Success.SubTitle') }}
+        </h2>
+        <p class="successDescription">
+          {{ $t('reportBug.Success.Description') }}
+        </p>
+        <Button
+          @click="resolve"
+        >
+          {{ $t('reportBug.Success.Return') }}
+        </Button>
+      </template>
+    </div>
+  </div>
+</template>
+
+<script>
+import { detect } from 'detect-browser';
+import FailureIcon from '../assets/APIError.svg?icon-component';
+import IconClose from '../assets/iconCloseRebranded.svg?icon-component';
+import IconEye from '../assets/iconEye.svg?icon-component';
+import IconOK from '../assets/iconOK.svg?icon-component';
+import ButtonPlain from './ButtonPlain.vue';
+import { issueReport } from '../utils/backend';
+import Button from './Button.vue';
+
+export default {
+  components: {
+    FailureIcon,
+    IconClose,
+    IconEye,
+    IconOK,
+    ButtonPlain,
+    Button,
+  },
+  props: {
+    resolve: { type: Function, required: true },
+    error: { type: [Object, Error, PromiseRejectionEvent.reason], required: true },
+  },
+  data() {
+    return {
+      report: '',
+      step: 1,
```

Comment:
I would name it `isSent` or `isSuccess` having a boolean value instead, we don't have any multiple steps